app/code/Magento/Elasticsearch/SearchAdapter/Query/Builder/Sort.php

```diff
@@ -124,4 +136,39 @@ public function getSort(RequestInterface $request)
 
         return $sorts;
     }
+
+    /**
+     * Prepare sort by saleability.
+     *
+     * @return array
+     */
+    public function getSortBySaleability()
```

Comment:
introduce the public function. @sidolov Can you give us some advice about that ?